tests/mocks/trade.mock.js

```diff
@@ -0,0 +1,70 @@
+const tradesJSON = [
+  {
+    id: 1,
+    state: 'awaiting_transfer_in'
+  },
+  {
+    id: 2,
+    state: 'awaiting_transfer_in'
+  }
+];
+
+let publicTradesJSON = [];
+
+let init = () => {
+  publicTradesJSON.length = 0;
+  JSON.parse(JSON.stringify(tradesJSON)).forEach(tradeJSON => {
+    publicTradesJSON.push(tradeJSON);
+  });
+};
+
+init();
+
+class Trade {
```

Comment:
See my first comment. For unit tests, a good mock should be the bare minimum needed. The fact that this mock is so large is probably a sign that either `exchange.js` or `trade.js` needs more refactoring. Ideally we wouldn't need more than a few Jasmine spies to mock the `Trade` behavior. In addition to unit tests, it might be useful to add integration tests that do use the real object.

https://www.toptal.com/qa/how-to-write-testable-code-and-why-it-matters

I think our mocks should be something like the `IDateTimeProvider` example. Simple, but still its own class.